File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/command/CommandPattern.java
package jv.com.tencent.cain.design_pattern.command;

/**
 * @author cainjiang
 * @date 2020/7/20
 */
public class CommandPattern {

    public static void main(String[] args) {
        Receiver receiver = new Receiver();
        AbstractCommand command = new ConcrectCommand(receiver);
        Invoker invoker = new Invoker(command);
        invoker.executeCommand();
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/command/CommandPattern.kt
package kt.com.tencent.cain.desigin_pattern.command

/**
 * @author cainjiang
 * @date 2020/7/20
 */

fun main() {
    val receiver = Receiver()
    val command: BaseCommand = ConcrectCommand(receiver)
    val invoker = Invoker(command)
    invoker.executeCommand()
    println("Hello World.")
}